### PR TITLE
NIAD-1021: Translate Observation.interpretation additional unit tests

### PR DESCRIPTION
**Pull request by ElliotB-96**
* Implemented additional unit tests for more interpretation codes.

### CODE REVIEW

File: service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ObservationStatementMapperTest.java [hunk 1]
```diff
@@ -128,7 +146,7 @@ public void When_MappingObservationJson_Expect_ObservationStatementXmlOutput(Str
         Observation parsedObservation = new FhirParseService().parseResource(jsonInput, Observation.class);
 
         String outputMessage = observationStatementMapper.mapObservationToObservationStatement(parsedObservation, false);
-
+        System.out.println(outputMessage);
```
Comment (n-r-b) [inline service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ObservationStatementMapperTest.java:149]: Can this be removed?

File: service/src/test/resources/ehr/mapper/observation/example-observation-resource-21.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  ],
+  "code": {
+    "text": "test"
+  },
+  "interpretation": {
+    "coding": [
+      {
+        "system": "http://hl7.org/fhir/v2/0078",
+        "code": "LU",
+        "display": "Very High Value"
```
Comment (n-r-b) [inline service/src/test/resources/ehr/mapper/observation/example-observation-resource-21.json:23]: Can you update this to reflect the test scenario?

File: service/src/test/resources/ehr/mapper/observation/example-observation-resource-19.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  ],
+  "code": {
+    "text": "test"
+  },
+  "interpretation": {
+    "coding": [
+      {
+        "system": "http://hl7.org/fhir/v2/0078",
+        "code": "L",
+        "display": "Very High Value"
```
Comment (n-r-b) [inline service/src/test/resources/ehr/mapper/observation/example-observation-resource-19.json:23]: Can you also update this to reflect the test scenario?

File: service/src/test/resources/ehr/mapper/observation/example-observation-resource-20.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  ],
+  "code": {
+    "text": "test"
+  },
+  "interpretation": {
+    "coding": [
+      {
+        "system": "http://hl7.org/fhir/v2/0078",
+        "code": "LL",
+        "display": "Very High Value"
```
Comment (n-r-b) [inline service/src/test/resources/ehr/mapper/observation/example-observation-resource-20.json:23]: And this one too

File: service/src/test/resources/ehr/mapper/observation/example-observation-resource-23.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  ],
+  "code": {
+    "text": "test"
+  },
+  "interpretation": {
+    "coding": [
+      {
+        "system": "http://hl7.org/fhir/v2/0078",
+        "code": "AA",
+        "display": "Very High Value"
```
Comment (n-r-b) [inline service/src/test/resources/ehr/mapper/observation/example-observation-resource-23.json:23]: Can you update this to reflect the test scenario?

File: service/src/test/resources/ehr/mapper/observation/example-observation-resource-22.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  ],
+  "code": {
+    "text": "test"
+  },
+  "interpretation": {
+    "coding": [
+      {
+        "system": "http://hl7.org/fhir/v2/0078",
+        "code": "A",
+        "display": "Very High Value"
```
Comment (n-r-b) [inline service/src/test/resources/ehr/mapper/observation/example-observation-resource-22.json:23]: And this one too

File: service/src/test/resources/ehr/mapper/observation/expected-output-observation-statement-16.xml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+        <statusCode code="COMPLETE" />
+        <effectiveTime>
+            <center value="20190601113000"/>
+        </effectiveTime>
+        <availabilityTime value="20190601113000"/>
+        
+        <pertinentInformation typeCode="PERT">
+            <sequenceNumber value="+1" />
+            <pertinentAnnotation classCode="OBS" moodCode="EVN">
+                <text>Range: Text: Test reference range text High: 20 Test comment</text>
```
Comment (n-r-b) [inline service/src/test/resources/ehr/mapper/observation/expected-output-observation-statement-16.xml:13]: Should this mention "High" when the interpretation code is low (that's if this is related)? I'm not too familiar with the logic behind this

**Review by mattd-kainos — CHANGES_REQUESTED** [review]
@ElliotB-96  to double check that all four conditions described by the mapping spreadsheet have been covered

**Review by mattd-kainos — DISMISSED** [review]
@ElliotB-96 provided evidence interpretation text is covered

**Review by n-r-b — DISMISSED** [review]

**Review by Broom94 — DISMISSED** [review]

File: service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ObservationStatementMapperTest.java [hunk 8]
```diff
@@ -132,7 +150,7 @@ public void When_MappingObservationJson_Expect_ObservationStatementXmlOutput(Str
...
```
Comment (n-r-b) [inline service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ObservationStatementMapperTest.java:153]: It's re-appeared again!

**Review by n-r-b — APPROVED** [review]

**Review by mattd-kainos — APPROVED** [review]

**Pull request merged**